src/appleseed/renderer/modeling/material/material.cpp

```diff
@@ -228,32 +228,25 @@ IBasisModifier* Material::create_basis_modifier(const MessageContext& context) c
     const Source* displacement_source = m_inputs.source("displacement_map");
 
     // Nothing to do if there is no displacement source.
-    if (displacement_source == 0)
-        return 0;
-
-    // Only texture instances can be bound to the displacement map input.
-    if (dynamic_cast<const TextureSource*>(displacement_source) == 0)
-    {
-        RENDERER_LOG_ERROR(
-            "%s: a texture instance must be bound to the \"displacement_map\" input.",
-            context.get());
-        return 0;
-    }
-
-    // Retrieve the displacement texture.
-    const TextureSource* displacement_map = static_cast<const TextureSource*>(displacement_source);
-    const Texture& texture = displacement_map->get_texture_instance().get_texture();
+    if (displacement_source == nullptr)
+        return nullptr;
 
-    // Print a warning if the displacement texture is not expressed in the linear RGB color space.
-    if (texture.get_color_space() != ColorSpaceLinearRGB)
+    // Additional checks if a texture instance is used for displacement.
+    const TextureSource* texture_map = dynamic_cast<const TextureSource*>(displacement_source);
```

Comment:
Franz, I think it is still looking for TextureSource here?

Reply:
Yes that's correct, it's checking if the source is a texture source and if so, does an additional check.

Reply:
Nevermind, it's something else. I'm still checking.

Reply:
Something is wrong, still checking..:
![image](https://user-images.githubusercontent.com/1850877/30561926-b57f2874-9c8a-11e7-827e-6943691ca801.png)

Reply:
I see. In the appleseed code, try decommenting the line that says

    //#define USE_SCREEN_SPACE_UV_DERIVATIVES

on line 53 of `src\appleseed\renderer\modeling\material\bumpmappingmodifier.cpp`.

Reply:
That's an alternative bump mapping method that may generally work better.